### PR TITLE
re-organize gitops cmds to dedup codes

### PR DESCRIPTION
**Pull request by chanwit**
This PR re-organizes `gitops` sub-commands to 

- remove duplicated codes #2946
- fix persistent flags problems

Signed-off-by: Chanwit Kaewkasi <chanwit@weave.works>

### CODE REVIEW

File: cmd/gitops/config/options.go [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-	GitHostTypes          map[string]string
-	InsecureSkipTLSVerify bool
-	Username              string
-	Password              string
-	Kubeconfig            string
+	Endpoint          string
+	OverrideInCluster bool
+	GitHostTypes      map[string]string
+	Username          string
+	Password          string
```
Comment (ozamosi) [inline cmd/gitops/config/options.go:8]: I think these all _should_ really be in enterprise - except username and password is reused in the dashboard install command.

File: cmd/gitops/root/root.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	"k8s.io/client-go/rest"
+)
+
+const defaultNamespace = "flux-system"
+
+var (
+	rootCmdFlags      = &config.Options{}
+	kubeconfigArgs    = genericclioptions.NewConfigFlags(false)
+	kubeclientOptions = new(runclient.Options)
+)
```
Comment (ozamosi) [inline cmd/gitops/root/root.go:26]: While this is very sleek, I think this won't work easily with enterprise right now, as it's just a copy of this file. I don't think it's insurmountable, but they _will_ be in another package.

**Review by ozamosi — COMMENTED** [review]
:tada: I really like the general direction this is taking.

**Comment by bigkevmcd** [comment]
@chanwit I'm going to close this just now, feel free to reopen if you want to pick it back up

**Pull request closed without merging**